### PR TITLE
rationalisation of logs and removal of duplicate codes (it should fix also #931) 

### PR DESCRIPTION
**Pull request by Odyno**
## Description:
Rationalisation of logs with removal of duplicate codes (it should fix also #931) 

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] I accept the [DCO](https://github.com/1technophile/OpenMQTTGateway/blob/development/docs/participate/development.md#developer-certificate-of-origin).

### CODE REVIEW

File: main/main.ino [hunk 1]
```diff
@@ -286,15 +286,123 @@ bool to_bool(String const& s) { // thanks Chris Jester-Young from stackoverflow
...
+ * @return char* the string
+ */
+const char* to_char(JsonObject& jsondata) {
+  String out = "";
+  serializeJson(jsondata, out);
+  return out.c_str();
+}
+
+/**
+ * @brief from unsigned long to chart *
```
Comment (1technophile) [inline main/main.ino]: small typo chart -> char

Comment (Odyno) [reply main/main.ino]: Hi! I have discovered that this code is not thread-safe and contain a variable scope problem, I am updating the pr just now

**Comment by 1technophile** [comment]
One next step could be to rework the pubMQTT calls leveraging the toChar functions added.

**Comment by Odyno** [comment]
Unfortunately the problem is in the to_char because: if I put the prefix "static char xxx[yy]" they are not thread safe, but if I don't put it they are subject to the stack clean when the function exits.

**Comment by Odyno** [comment]
@1technophile  I've updated the code but had to put it back inline. I agree with you to make a review of the 30 or so versions of pubMQTT, maybe it would be useful to create an ad hoc class (maybe an extension of "client") that also manages the connection.

File: main/config_mqttDiscovery.h [hunk 2]
```diff
@@ -94,7 +94,7 @@ void announceDeviceTrigger(bool use_gateway_info,
 #define DEVICEMANUFACTURER "OMG_community"
 
 /*-------------- Auto discovery macros-----------------*/
-// Set the line below to true so as to have autodiscovery working with OpenHAB
+// Set the line below to true so as to have auto working with OpenHAB
```
Comment (1technophile) [inline main/config_mqttDiscovery.h]: Not sure this removal is necessary?

File: main/ZsensorDS1820.ino [hunk 3]
```diff
@@ -102,7 +102,7 @@ void pubOneWire_HADiscovery() {
 #  ifdef ZmqttDiscovery
   // If zmqtt discovery is enabled, create a sensor topic for each DS18b20 sensor found on the bus, using addr as uniqueID
   if (disc) {
-    Log.notice(F("CreateDiscoverySensor - Found %d" CR), ds1820_count);
+    //Log.notice(F("CreateDiscoverySensor - Found %d" CR), ds1820_count);
```
Comment (1technophile) [inline main/ZsensorDS1820.ino]: Maybe we can remove the line if we want to be consistent with the others removals

**Comment by 1technophile** [comment]
> maybe it would be useful to create an ad hoc class (maybe an extension of "client") that also manages the connection.

Indeed It would be interesting to have something more modular and that can be extended to further protocols

File: main/ZsensorDS1820.ino (hunk 3 above)
Comment (Odyno) [reply main/ZsensorDS1820.ino]: ok

File: main/config_mqttDiscovery.h (hunk 2 above)
Comment (Odyno) [reply main/config_mqttDiscovery.h]: true

**Comment by 1technophile** [comment]
Could you rebase it from development please, and I think we are good to go

**Comment by Odyno** [comment]
@1technophile  I have a problem now... the current *development*  branch for me is out of size

```
....
Linking .pio/build/az-delivery/firmware.elf
Retrieving maximum program size .pio/build/az-delivery/firmware.elf
Checking size .pio/build/az-delivery/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
Error: The program size (1479456 bytes) is greater than maximum allowed (1310720 bytes)
RAM:   [=         ]  11.7% (used 62276 bytes from 532480 bytes)
Flash: [===*** [checkprogsize] Explicit exit, status 1
=======]  112.9% (used 1479456 bytes from 1310720 bytes)
============================================================ [FAILED] Took 51.47 seconds ============================================================

Environment    Status    Duration
-------------  --------  ------------
az-delivery    FAILED    00:00:51.469
======================================================= 1 failed, 0 succeeded in 00:00:51.469 =======================================================
The terminal process "platformio 'run'" terminated with exit code: 1.
```

I have this configuration
```
[platformio]
default_envs = az-delivery

[env:az-delivery]
lib_ldf_mode =  deep+
platform = espressif32
board = az-delivery-devkit-v4
check_tool = cppcheck, clangtidy
check_flags =
  cppcheck: -v
  clangtidy: -fix-errors -format-style=mozilla

lib_deps =
  ${env:esp32dev-all-test.lib_deps}
build_flags =
  ${env:esp32dev-all-test.build_flags}
  '-DZgatewayBT="BT"'
  '-DZgatewayRF="RF"'
  '-DLED_RECEIVE=2'
  '-DLED_RECEIVE_ON=0'
  '-DpubUnknownBLEServiceData=false'
  '-DsimpleReceiving=true'
  '-DZmqttDiscovery="HADiscovery"'
  '-DGateway_Name="OMG_BTRF_Latest"'
  '-DESPWifiManualSetup=true'
  '-Dwifi_ssid="+++++"'
  '-Dwifi_password="+++++"'
  '-DMQTT_PORT="+++++"'
  '-DMQTT_USER="+++++"'
  '-DMQTT_PASS="+++++"'
  '-DMQTT_SERVER="+++++"'
  #'-DOMG_VERSION="0.9.6-4"'
  '-DLOG_LEVEL=LOG_LEVEL_VERBOSE'  ; Enable trace level logging
  '-DvalueAsASubject=true'

```

do you have suggestion?

**Comment by 1technophile** [comment]
>do you have suggestion?

Maybe by identifying the difference with this [one](https://github.com/1technophile/OpenMQTTGateway/blob/10f0d4cff488ef68e9e874c2bee9518ea5174d35/platformio.ini#L202),
one issue also could be that Arduino_json is duplicated into the binary due to the call from [decoder library](https://github.com/theengs/decoder/tree/development/src) and the direct call to it.
I have tried to avoid it but it may be the case

**Comment by Odyno** [comment]
@1technophile  fixed, just used   ${com-esp.build_flags} instead of    ${env:esp32dev-all-test.build_flags}

**Comment by 1technophile** [comment]
Thanks

**Pull request merged**